lib/rubocop/cop/lint/unneeded_require_statement.rb

```diff
@@ -0,0 +1,50 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Lint
+      # Checks for unnecessary `require` statement.
+      #
+      # The following features are unnecessary `require` statement because
+      # they are already loaded.
+      #
+      # ruby -ve 'p $LOADED_FEATURES.reject { |feature| %r|/| =~ feature }'
+      # ruby 2.2.8p477 (2017-09-14 revision 59906) [x86_64-darwin13]
+      # ["enumerator.so", "rational.so", "complex.so", "thread.rb"]
+      #
+      # This cop targets Ruby 2.2 or higher containing these 4 features.
+      #
+      # @example
+      #   # bad
+      #   require 'unloaded_feature'
+      #   require 'thread'
+      #
+      #   # good
+      #   require 'unloaded_feature'
+      class UnneededRequireStatement < Cop
+        extend TargetRubyVersion
+
+        minimum_target_ruby_version 2.2
+
+        MSG = 'Remove unnecessary `require` statement.'.freeze
+
+        def_node_matcher :unnecessary_require_statement?, <<-PATTERN
+          (send nil :require
+            (str {"enumerator" "rational" "complex" "thread"}))
```

Comment:
Should this be dynamically created based on the running version of Ruby? Ruby 2.0 has `enumerator`, but not the others.

Reply:
This is based on Ruby 2.2.8. And these 4 features including `enumerator` are the same from Ruby 2.2 to the current Ruby 2.5-dev.
https://github.com/bbatsov/rubocop/pull/4764#discussion_r139868696